packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -519,7 +519,8 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                         property.locationOf()
                                     )
                                 }
-                                val isIndexed = backingField.hasAnnotation(INDEX_ANNOTATION)
+                                // See https://github.com/realm/realm-core/issues/6187
```

Comment:
Hmm, right now I'm adding it here,  while waiting for the Core fix, but also because I'm a little afraid that Core might not do it correctly at all, so just hacking the return value would just seem to mask even more bugs? 🤔

Reply:
After discussion, I'll remove this fix for now and revert the test. This is now a "known bug" and we will talk with Core about a resolution.